packages/cli/src/ui/components/shared/TextInput.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type React from 'react';
+import { useState, useEffect, useCallback } from 'react';
+import type { Key } from '../../hooks/useKeypress.js';
+import { Text, Box } from 'ink';
+import { useKeypress } from '../../hooks/useKeypress.js';
+import chalk from 'chalk';
+import { theme } from '../../semantic-colors.js';
+
+export interface TextInputProps {
+  value: string;
+  placeholder?: string;
+  onChange: (value: string) => void;
+  onSubmit?: (value: string) => void;
+  onCancel?: () => void;
+  mask?: string;
+  focus?: boolean;
+}
+
+export function TextInput({
+  value,
+  placeholder = '',
+  onChange,
+  onSubmit,
+  onCancel,
+  mask,
+  focus = true,
+}: TextInputProps): React.JSX.Element {
+  const [cursorOffset, setCursorOffset] = useState(value.length);
+  const [isExiting, setIsExiting] = useState(false);
+
+  useEffect(() => {
+    // When the controlled value changes, update the cursor position if it's
+    // out of bounds. This can happen if the parent component changes the
+    // value programmatically.
+    if (cursorOffset > value.length) {
+      setCursorOffset(value.length);
+    }
+  }, [value, cursorOffset]);
+
+  useEffect(() => {
+    if (isExiting && onCancel) {
+      // Give Ink a moment to run the useInput cleanup before we trigger the
```

Comment:
this seems obsolete now that we aren't using useInput. Would guess this is a hack to try to fix odd bugs in useInput when you were using it